pkgs/development/python-modules/mcpo/default.nix

```
@@ -0,0 +1,67 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  hatchling,
+  click,
+  fastapi,
+  mcp,
+  passlib,
+  pydantic,
+  pyjwt,
+  pytest,
+  python-dotenv,
+  pythonOlder,
+  typer,
+  uvicorn,
+}:
+
+buildPythonPackage rec {
+  pname = "mcpo";
+  version = "0.0.14";
+  pyproject = true;
+
+  disabled = pythonOlder "3.11";
+
+  src = fetchFromGitHub {
+    owner = "open-webui";
+    repo = pname;
+    tag = "v${version}";
+    hash = "sha256-viwf0wjNvp/MyJHHYrWOqQe/bUJp6y7Yu9uVan7gC/U=";
+  };
+
+  build-system = [
+    hatchling
+  ];
+
+  nativeCheckInputs = [
+    pytest
```

We have a hook to avoid invoking `pytest` manually in the checkPhase
```suggestion
    pytestCheckHook
```
You may also want to move `nativeCheckInputs` under dependencies